lib/benchmark/sweet.rb

```diff
@@ -103,10 +107,20 @@ def self.to_table(arr)
 
       printf format, *column_sizes.collect { |w| "-" * w }
 
-      arr[0..arr.count].each do |line|
+      arr.each do |line|
         printf format, *line.each_with_index.map { |el, i| " "*(column_sizes[i] - field_sizes[line][i] ) + el[1].to_s }
       end
     end
+
+    def self.normalize_data(rows, cols)
```

Comment:
So I don't have a full solution for you here, but this is something to think about:

What if the given number of column names are equal, but have different labels.  For example, if I changed a section of your example to this:

```diff
   x.metadata data: 'FULL' do
-    x.report("x.blank?")           { FULL.blank? }
+    x.report("x is blank")         { FULL.blank? }
     x.report("x.empty?")           { FULL.empty? }
   end
```

You end up getting an error "too few arguments", even with your code change.

* * *

I did the following on my end, and it seems to have a "safer" result, albeit requires two loops through `rows` to do it:

```ruby
    def self.normalize_data(rows, cols)
      keys = []
      rows.each { |r| keys |= r.keys }
      rows.each { |r| keys.each { |k| r[k] ||= nil } }
      rows
    end
```

But instead of trying to get a "count" of the keys, it gets a master list, and then makes sure to assign a value for each key, regardless of row when normalizing.

This still has troubles, since the output instead looks like this:

```
    method |                    NIL |                  EMPTY | FULL
---------- | ---------------------- | ---------------------- |
  x.blank? |         24901111.2 i/s | 23533921.1 i/s - 1.29x |
  x.empty? |         30325413.4 i/s |         30208258.1 i/s |
x is blank | 23933916.8 i/s - 1.26x |                        |
```

Which isn't right, but at least doesn't error.  Still seems like it requires some fixes elsewhere to handle the column name discrepancy elsewhere.